### PR TITLE
[flytepropeller][compiler] Error Handling when Type is not found

### PR DESCRIPTION
**Pull request by Future-Outlier**
## Tracking issue
https://linear.app/unionai/issue/DX-948/flytepropeller-compiler-error-handling-when-type-is-not-found

## Why are the changes needed?
When developing `JSON IDL`, I found that if I registered task to flytepropeller but flytepropeller didn't  support the corresponding `flyteidl`, flyteadmin will crash and the error message from flyteadmin doesn't have a clear error message.

#### flytekit
image

#### flytepropeller
image

### After
We should give a good error message and avoid crashing the propeller.

image

## What changes were proposed in this pull request?
Improve error message when `inputType` is `nil`, which means flytepropeller didn't support that type.

## How was this patch tested?
single binary and unit test.

### Setup process
flytekit: https://github.com/flyteorg/flytekit/pull/2600
flyteidl: https://github.com/flyteorg/flyte/pull/5542
flyte: this branch
```python
from flytekit import task, workflow

@task()
def t5(input: dict) -> dict:
    print(input)
    print("different")
    return input.update({"b": 2}).update({"c": 3})

@workflow
def wf2(input: dict={'d': 1, }) -> dict:
    return t5(input=input)
```
```python
pyflyte run --remote build/PR/JSON/example.py wf2
```

### Screenshots
image

## Check all the applicable boxes 

- [x] I updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] All commits are signed-off.

## Related PRs
https://github.com/flyteorg/flyte/pull/5542

### CODE REVIEW

File: flyteadmin/pkg/manager/impl/validation/validation.go [hunk 1]
```diff
@@ -282,11 +283,26 @@ func validateParameterMap(inputMap *core.ParameterMap, fieldName string) error {
 			defaultValue := defaultInput.GetDefault()
 			if defaultValue != nil {
 				inputType := validators.LiteralTypeForLiteral(defaultValue)
+
+				if inputType == nil {
+					return errors.NewFlyteAdminErrorf(codes.InvalidArgument,
+						fmt.Sprintf(
+							"Flyte Propeller encountered an issue while determining\n"+
+								"the type of the default value for Parameter '%s' in '%s'.\n"+
+								"Registered type from FlyteKit: [%s].\n"+
```
Comment (pingsutw) [inline flyteadmin/pkg/manager/impl/validation/validation.go:292]: The user may use flytekit-java instead of flytekit.

**Comment by eapolinario** [comment]
@Future-Outlier , can you fix the [lint error](https://github.com/flyteorg/flyte/actions/runs/10193513817/job/28198271351?pr=5612)?

**Comment by wild-endeavor** [comment]
@Future-Outlier just wondering does this fail in the same way for pyflyte register?  or is it just run?

**Comment by Future-Outlier** [comment]
> @Future-Outlier just wondering does this fail in the same way for pyflyte register?  or is it just run?

Both failed
They will go through the same code piece

**Review by pingsutw — APPROVED** [review]

**Pull request merged**